tests/semanticdb/expect/Advanced.expect.scala

```diff
@@ -42,6 +42,10 @@ object Test/*<-advanced::Test.*/ {
         ()
     }
   }
+
+  // see: https://github.com/lampepfl/dotty/pull/14608#discussion_r835642563
+  lazy val foo/*<-advanced::Test.foo.*/: (reflect.Selectable/*->scala::reflect::Selectable#*/ { type A/*<-local16*/ = Int/*->scala::Int#*/ }) &/*->scala::`&`#*/ (reflect.Selectable/*->scala::reflect::Selectable#*/ { type A/*<-local17*/ = Int/*->scala::Int#*/; val a/*<-local18*/: A/*->local17*/ }) = ???/*->scala::Predef.`???`().*/
+  def bar/*<-advanced::Test.bar().*/: foo/*->advanced::Test.foo.*/.A = foo/*->advanced::Test.foo.*/.a
```

Comment:
I guess this is because `.A` and `.a` (access the field of refinements), don't have symbols (we're crafting fake symbols for them), and `ExtractSemanticDB` doesn't resolve the fake symbols by name.

We can see the same problem around here:
https://github.com/lampepfl/dotty/blob/60c336ef34fe0e77851a7652b910db13b84dfe6f/tests/semanticdb/expect/Advanced.expect.scala#L17

I'll take a look :eyes:

Reply:
we have to resolve the symbol from name here https://github.com/lampepfl/dotty/blob/60c336ef34fe0e77851a7652b910db13b84dfe6f/compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala#L263-L264

Reply:
I guess looking again `registerUse` or `registerUseGuarded` still only assume real symbols.

Reply:
As per `foo.A`, https://github.com/lampepfl/dotty/pull/14608/commits/b9646942facccdcc110753f019c14b941fae870d as you mentioned, you're right, we could fix the issue by looking up the fake symbols and `registerUse` them.

In terms of `foo.a` (field access to structural type), it is transformed to `foo.selectDynamic("a")` by Typer here ↓
https://github.com/lampepfl/dotty/blob/b9646942facccdcc110753f019c14b941fae870d/compiler/src/dotty/tools/dotc/typer/Dynamic.scala#L49-L65
We have to check if `qual` is something like `Apply(Select(Ident(foo), name), List(Literal(Constant("a")))) if name == nme.applyDynamic || name == nme.selectDynamic || name == nme.updateDynamic || name == nme.applyDynamicNamed`, and lookup symbol for `a` from `foo` (This is not yet fixed, can I work on this in another PR?)